### DISCUSSION TITLE
Use of `static` instead of `self` for better extensibility

### DISCUSSION THREAD

**Discussion by bilogic** [spatie/laravel-google-calendar#289] [Q&A]
1. We are extending the `GoogleCalendarFactory` class in order to write to multiple calendars instead of just 1 fixed calendar
2. However, most of the calls in the class uses `self::` instead of `static::`, thus, we had to overwrite several methods that look identical in text form
3. Possible to change from using `self::` to `static::`? I'm more than happy to file a PR

**Comment by freekmurze** [comment]
Feel free to PR this
